### PR TITLE
mix in the mixins ..., refactoring unittests

### PR DESCRIPTION
**Pull request by y0urself**
I will already create a PR for the Mixins-Idea, so that you can track the progress @bjoernricks 

So far this includes changes from:

* https://github.com/greenbone/gvmd/pull/790
* https://github.com/greenbone/gvmd/pull/922
* https://github.com/greenbone/gvmd/pull/1151

This should already be covered by `gmpv8`:
* https://github.com/greenbone/gvmd/pull/1116 
* https://github.com/greenbone/gvmd/pull/1121

This now contains all new test cases, refactored with Mixins. Fixed many tests, test-names and more.
I don't really can check if I tested all possible cases ... there are >4000 tests now! ...

**Checklist**:

- [x] Tests
- [ ] [CHANGELOG](https://github.com/greenbone/python-gvm/blob/master/CHANGELOG.md) Entry
- [ ] Documentation

### CODE REVIEW

**Comment by y0urself** [comment]
Sorry. This was like hell!

Updating the tests was really horror.
I think I destroyed some file history due to the requested naming changes by @bjoernricks

File: gvm/protocols/gmpv208/gmpv208.py [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+    def create_agent(
+        self,
+        installer: str,
+        signature: str,
+        name: str,
+        *,
+        comment: Optional[str] = None,
+        howto_install: Optional[str] = None,
+        howto_use: Optional[str] = None
+    ) -> Any:
```
Comment (bjoernricks) [inline gvm/protocols/gmpv208/gmpv208.py:74]:
```suggestion
    ) -> None:
```

File: gvm/protocols/gmpv208/gmpv208.py [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+        # pylint: disable=unused-argument
+        deprecation(
+            "{} is deprecated in version {}{}".format(
```
Comment (bjoernricks) [inline gvm/protocols/gmpv208/gmpv208.py:77]:
```suggestion
            "{} has been removed in GMP version {}{}".format(
```

File: gvm/protocols/gmpv208/gmpv208.py [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+                self.create_agent.__name__,
+                self.get_protocol_version()[0],
+                self.get_protocol_version()[1],
+            )
+        )
+
+    def clone_agent(self, agent_id: str) -> Any:
+        # pylint: disable=unused-argument
+        deprecation(
+            "{} is deprecated in version {}{}".format(
```
Comment (bjoernricks) [inline gvm/protocols/gmpv208/gmpv208.py:87]:
```suggestion
            "{} has been removed in GMP version {}{}".format(
```

File: gvm/protocols/gmpv208/gmpv208.py [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+    def modify_agent(
+        self,
+        agent_id: str,
+        *,
+        name: Optional[str] = None,
+        comment: Optional[str] = None
+    ) -> Any:
+        # pylint: disable=unused-argument
+        deprecation(
+            "{} is deprecated in version {}{}".format(
```
Comment (bjoernricks) [inline gvm/protocols/gmpv208/gmpv208.py:103]:
```suggestion
            "{} has been removed in GMP version {}{}".format(
```

File: gvm/protocols/gmpv208/gmpv208.py [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+            )
+        )
+
+    def modify_agent(
+        self,
+        agent_id: str,
+        *,
+        name: Optional[str] = None,
+        comment: Optional[str] = None
+    ) -> Any:
```
Comment (bjoernricks) [inline gvm/protocols/gmpv208/gmpv208.py:100]:
```suggestion
    ) -> None:
```

File: gvm/protocols/gmpv208/gmpv208.py [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+        # pylint: disable=unused-argument
+        deprecation(
+            "{} is deprecated in version {}{}".format(
+                self.create_agent.__name__,
+                self.get_protocol_version()[0],
+                self.get_protocol_version()[1],
+            )
+        )
+
+    def clone_agent(self, agent_id: str) -> Any:
```
Comment (bjoernricks) [inline gvm/protocols/gmpv208/gmpv208.py:84]:
```suggestion
    def clone_agent(self, agent_id: str) -> None:
```

File: gvm/protocols/gmpv208/gmpv208.py [hunk 7]
```diff
@@ -0,0 +1,358 @@
...
+            )
+        )
+
+    def delete_agent(
+        self,
+        agent_id: str,
+        *,
+        ultimate: Optional[bool] = False
+        # pylint: disable=unused-argument
+    ) -> Any:
```
Comment (bjoernricks) [inline gvm/protocols/gmpv208/gmpv208.py:116]:
```suggestion
    ) -> None:
```

File: gvm/protocols/gmpv208/gmpv208.py [hunk 8]
```diff
@@ -0,0 +1,358 @@
...
+        deprecation(
+            "{} is deprecated in version {}{}".format(
```
Comment (bjoernricks) [inline gvm/protocols/gmpv208/gmpv208.py:118]:
```suggestion
            "{} has been removed in GMP version {}{}".format(
```

File: gvm/protocols/gmpv208/gmpv208.py [hunk 9]
```diff
@@ -0,0 +1,358 @@
...
+                self.delete_agent.__name__,
+                self.get_protocol_version()[0],
+                self.get_protocol_version()[1],
+            )
+        )
+
+    def verify_agent(self, agent_id: str) -> Any:
```
Comment (bjoernricks) [inline gvm/protocols/gmpv208/gmpv208.py:125]:
```suggestion
    def verify_agent(self, agent_id: str) -> None:
```

File: gvm/protocols/gmpv208/gmpv208.py [hunk 10]
```diff
@@ -0,0 +1,358 @@
...
+        # pylint: disable=unused-argument
+        deprecation(
+            "{} is deprecated in version {}{}".format(
```
Comment (bjoernricks) [inline gvm/protocols/gmpv208/gmpv208.py:128]:
```suggestion
            "{} has been removed in GMP version {}{}".format(
```

File: gvm/protocols/gmpv208/gmpv208.py [hunk 11]
```diff
@@ -0,0 +1,358 @@
...
+                self.verify_agent.__name__,
+                self.get_protocol_version()[0],
+                self.get_protocol_version()[1],
+            )
+        )
+
+    def get_agent(self, agent_id: str) -> Any:
```
Comment (bjoernricks) [inline gvm/protocols/gmpv208/gmpv208.py:135]:
```suggestion
    def get_agent(self, agent_id: str) -> None:
```

File: gvm/protocols/gmpv208/gmpv208.py [hunk 12]
```diff
@@ -0,0 +1,358 @@
...
+        # pylint: disable=unused-argument
+        deprecation(
+            "{} is deprecated in version {}{}".format(
```
Comment (bjoernricks) [inline gvm/protocols/gmpv208/gmpv208.py:138]:
```suggestion
            "{} has been removed in GMP version {}{}".format(
```

**Pull request merged**